### PR TITLE
codewhisperer: patch java test file regex pattern MyClassTests.java

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Before
- TestMyClass.java
- MyClassTest.java

## After
- TestMyClass.java
- MyClassTest.java
- MyClassTests.java

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/codewhisperer/util/supplementalContext/codeParsingUtil.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export const utgLanguageConfigs: Record<string, utgLanguageConfig> = {
     // Java regexes are not working efficiently for class or function extraction
     java: {
         extension: '.java',
-        testFilenamePattern: /(?:Test([^/\\]+)\.java|([^/\\]+)Test\.java)$/,
+        testFilenamePattern: /(?:Test([^/\\]+)\.java|([^/\\]+)Test\.java|([^/\\]+)Tests\.java)$/,
```
Comment (Will-ShaoHua) [inline src/codewhisperer/util/supplementalContext/codeParsingUtil.ts:22]: I still prefer a list of smaller regex like https://github.com/aws/aws-toolkit-jetbrains/blob/main/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/JavaCodeWhispererFileCrawler.kt#L25, which is easier to work with. But I will take this as follow up and don't want to bring too much changes at one shot. Any suggestion on which one is better? @justinmk3 @andrewyuq

Comment (justinmk3) [reply src/codewhisperer/util/supplementalContext/codeParsingUtil.ts:22]: For lists of "or" (`|`) cases, a list of regexes is usually a good idea. Especially if the list will continue to grow.

**Review by justinmk3 — APPROVED** [review]
Does this have a customer-facing impact worth mentioning in the changelog?

File: src/codewhisperer/util/supplementalContext/codeParsingUtil.ts (hunk 1 above)
Comment (Will-ShaoHua) [reply src/codewhisperer/util/supplementalContext/codeParsingUtil.ts:22]: yea agree, it's quite error prone when working with this giant regex. I will take refactoring as a follow up.

Comment (andrewyuq) [reply src/codewhisperer/util/supplementalContext/codeParsingUtil.ts:22]: TODO comment?
and it looks like these two can be combined?
`([^/\\]+)Test\.java|([^/\\]+)Tests\.java`

Do we want to match aggressively or preservably? If it's the former do want to include any filename that contains the case-insensitive string `test`?

File: .changes/next-release/Bug Fix-d47847cc-bbb7-4c26-a102-82943152654a.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Bug Fix",
+	"description": "CodeWhisperer: patch java test file name MyClassTests.java will not be captured if file is not living in a test folder"
```
Comment (justinmk3) [inline .changes/next-release/Bug Fix-d47847cc-bbb7-4c26-a102-82943152654a.json:3]: Do customers know what "patch java test file" means?

```suggestion
	"description": "CodeWhisperer: java test file name MyClassTests.java will not be captured if file is not living in a test folder"
```

File: src/codewhisperer/util/supplementalContext/codeParsingUtil.ts (hunk 1 above)
Comment (Will-ShaoHua) [reply src/codewhisperer/util/supplementalContext/codeParsingUtil.ts:22]: I prefer not combining these 2 regex, I think it's much error prone. The final format should be more similar to JB's implementation which use a list of small regex, it's also for readability.

Comment (Will-ShaoHua) [reply src/codewhisperer/util/supplementalContext/codeParsingUtil.ts:22]: I think case sensitive is fine, let's do it case sensitive for now. We can update it if needed.

File: .changes/next-release/Bug Fix-d47847cc-bbb7-4c26-a102-82943152654a.json (hunk 2 above)
Comment (Will-ShaoHua) [reply .changes/next-release/Bug Fix-d47847cc-bbb7-4c26-a102-82943152654a.json:3]: sgg

**Review by justinmk3 — APPROVED** [review]

**Review by andrewyuq — APPROVED** [review]

**Pull request merged**